.circleci/config.yml

```diff
@@ -1,6 +1,19 @@
 version: 2.1
 
 jobs:
+  controller-test:
+    machine:
+      image: ubuntu-1604:201903-01
+    steps:
+      - checkout
+      - run:
+          name: Run Docker build
+          command: |
+            docker build -f controller/Dockerfile -t controller .
+      - run:
+          name: Run tests
+          command: |
+            docker run --privileged -t controller
```

Comment:
This really is "controller tests" rather than "controller", right? Would you mind renaming files accordingly?

Reply:
Technically, the build runs the tests. If the tests were to fail, and throw a non-zero exit code at the time of build, the `docker build` command above wouldn't succeed. So I don't think we need lines 13 onward at all?

Reply:
@jkff I'm not sure I understand. Are you referring the tag `controller`?

Reply:
> 
> 
> Technically, the build runs the tests. If the tests were to fail, and throw a non-zero exit code at the time of build, the `docker build` command above wouldn't succeed. So I don't think we need lines 13 onward at all?

I started with that, but it fails because LeakSanitizer, which runs during the tests, requires `ptrace`. This apparently requires the container be run in privileged mode. This is why the build step is separate from the test run step.